### PR TITLE
Correct GtkKeyHelper key codes for the Meta key

### PR DESCRIPTION
**Pull request by OmarEmaraDev**
## Description

While other key helpers associate the `metaModifier` key with the Super
key on Linux. The `GtkKeyHelper` associate it with the `GDK_META_MASK` mask,
the `GDK_KEY_Meta_L`, and `GDK_KEY_Meta_R` keys. Aside from the incorrect
handling of the keys, this also causes the assert in `_handleKeyEvent` to
fail upon pressing the Super key in a GTK flutter application.

This patch corrects the key codes and masks in the `GtkKeyHelper` to
correctly associate the metaModifier with the Super key to be in-line
with the other key helpers.

From `gdkkeysyms.h`:
```
  #define GDK_KEY_Super_L 0xffeb
  #define GDK_KEY_Super_R 0xffec
```
And from `gdktypes.h`:
```
  GDK_SUPER_MASK    = 1 << 26,
```
## Related Issues

This is similar to the following except it solves the issue with the Super key.

- #63939
- #68713

## Tests

I updated the following tests:

- raw_keyboard_test

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

- [x] Yes, this is a breaking change.
   - raw_keyboard_test needed to be updated, but it isn't a breaking change. 

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/services/raw_keyboard_linux.dart
Comment (robert-ancell) [inline packages/flutter/lib/src/services/raw_keyboard_linux.dart:359]: This list is in numerical order, so these should be moved below `altRightKeyCode`.

**Comment by robert-ancell** [comment]
Ugh, the meta/super naming is very confusing. I'm not an expert but reading https://askubuntu.com/questions/19558/what-are-the-meta-super-and-hyper-keys shows I did make a mistake and it should have been the super key.

Thanks @OmarEmaraDev! With the above ordering fix this LGTM.

File: packages/flutter/lib/src/services/raw_keyboard_linux.dart
Comment (OmarEmaraDev) [reply packages/flutter/lib/src/services/raw_keyboard_linux.dart]: @robert-ancell Done. Is the failing `flutter-build` something I should be worried about? (Seems to be failing for other pull requests as well)

Comment (robert-ancell) [reply packages/flutter/lib/src/services/raw_keyboard_linux.dart]: The failing flutter-build is the previous merge has caused a test failure. We just have to wait for a fix for that before this can be merged.

**Review by robert-ancell — APPROVED** [review]

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

Is the "meta" key just not used on Linux then (and the Windows/Command key is just called "super" instead)?

**Comment by robert-ancell** [comment]
@gspencergoog it seems that way. I've always heard the "Windows" key referred to as super on Linux desktops.

**Comment by goderbauer** [comment]
@robert-ancell Is this one ready to be submitted? If so, maybe tag it with "waiting for tree to go green" so it goes in when the tree is green?

**Comment by gspencergoog** [comment]
Sorry, I just realized that @robert-ancell wasn't the author of this (for some reason I read it as his PR). Submitting as soon as we get a green build.

**Pull request merged**
